packages/react/src/Banner/Banner.test.tsx

```diff
@@ -0,0 +1,211 @@
+import {render, screen} from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import React from 'react'
+import {Banner} from '../Banner'
+
+describe('Banner', () => {
+  it('should render as a region element', () => {
+    render(<Banner title="test" />)
+    expect(screen.getByRole('region', {name: 'test'})).toBeInTheDocument()
+  })
+
+  it('should label the landmark element with the title prop', () => {
+    render(<Banner title="test" />)
+    expect(screen.getByRole('region')).toEqual(screen.getByLabelText('test'))
+  })
+
+  it('should default the title to a h2', () => {
```

Comment:
I am a bit worried that having a default heading level will result in banners being added with incorrect heading hierarchy. I worry most people will set `title` without making the connection that it render a heading with a level that should be adjusted for their usecase.

What do you think?

Reply:
Makes total sense to me! Would the preference then be for this to always be an explicit heading level that the user of the component provides?

Reply:
Yes, I think that would be the most sure-way to encourage consumers to be mindful about the heading level and have proper heading hiearchy. I can't think of why we wouldn't want to require it, but let me know if there are concerns I'm not thinking of.

Another way to approach this would be to extend this lint rule - [a11y-explicit-heading](https://github.com/primer/eslint-plugin-primer-react/blob/main/src/rules/a11y-explicit-heading.js).

Reply:
Sounds good to me 👍 I think the only thing that comes up for this is that naming the prop on `Banner` is awkward (compared to on `Banner.Title`). 

Here are some options that came to mind, curious if any of them resonate or if there is a pattern that you would recommend following for this scenario:

- `titleHeadingLevel`
- `titleHeading`
- `heading`
- `headingLevel`

These could be props that accept a number (2-6) or the explicit tag (h2-h6).

Reply:
`titleAs` could definitely work! I also don't mind restricting these since `as` is technically for _any_ element and we really do want these specifically to only be headings in a specific range. Super torn on this, but if I had to pick I guess I would like:

```tsx
<Banner headingLevel={2}></Banner>

<Banner>
  <Banner.Title headingLevel={2}
</Banner>
```

Curious what you think

Reply:
I like that! That's a great point that `as` is technically for all any elements, and I do like the idea of moving away from it.